packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -168,6 +168,7 @@ export class CallContext {
   public setCallId(newCallId: string, oldCallId: string): void {
     this.modifyState((draft: CallClientState) => {
       const call = draft.calls[oldCallId];
+      call.id = newCallId;
```

Comment:
So this means the Call object that we get may not be the most updated call object and we want to apply on the correct Id before we set the Call object to the correct key?

Reply:
The id value in state.call.id has never been updated before this fix... Although we only use it in a very extreme case

Since this will be run at for the first callIdChange event, so this code will execute ahead of all the other state change, we should be good to assume customer will always get the right value in state